File: java/src/test/java/code/book/숫자카드게임/solution/P1Test.java
package code.book.숫자카드게임.solution;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class P1Test {

    @Test
    public void solution() {
        int[] n = {3,2};
        int[] m = {3,4};
        int[][][] list = {
            {
                {3,1,2},
                {4,1,4},
                {2,2,2}
            },
            {
                {7,3,1,8},
                {3,3,3,4}
            }
        };
        int[] ans = {2,3};

        for (int i = 0; i < n.length; i++)
            assertEquals(new P1().solutin(n[i], m[i], list[i]), ans[i]);
    }
}

File: codes/javascript/src/acmicpc/p1021/solution/p1.test.js
const solution = (n, m, list) => {

  // 간단하게 덱을 이용하는 방법이 있다.
  // 우선 2가지 방법이 있는데, 왼쪽 오른쪽, 비교해서 작은 쪽으로 이동하면 될듯

};

test([
  [10, 3, [1, 2, 3], 0],
  [10, 3, [2, 9, 5], 8],
  [32, 6, [27, 16, 30, 11, 6, 23], 59],
  [10, 10, [1, 6, 3, 2, 7, 9, 8, 4, 10, 5], 14],
])('회전하는 큐', (n, m, list, ans) => {
  expect(solution(n, m, list)).toBe(ans);
});